storage/series/series.go

```diff
@@ -440,6 +442,31 @@ func (s *dbSeries) mergeBlock(
 	return nil
 }
 
+func (s *dbSeries) Update(blk block.DatabaseBlock) error {
+	s.Lock()
+	defer s.Unlock()
+
+	// ensure series is bootstraped before allowing any updates
+	if s.bs != bootstrapped {
+		return errSeriesNotBootstrapped
+	}
+
+	// we only accept updates to non-buffered data
+	// ensure buffer is past provided time
+	min, _ := s.buffer.MinMax()
+	if !blk.StartTime().Before(min) {
```

Comment:
Ah I see, seems the min timestamp is actually the start of the past 2 hour block (not the current 2 hour block), doesn't that we won't repair the past 2 hour block?